Suspend with reminder alerts

ps2 opened the pull request:
https://tidepool.atlassian.net/browse/LOOP-1223

mpangburn commented on DashKitUI/Views/DashSettingsView.swift:
```diff
@@ -275,6 +280,32 @@ struct DashSettingsView<Model>: View where Model: DashSettingsViewModelProtocol
         ])
     }
+
+    var suspendOptionsActionSheet: ActionSheet {
```
[nit] Might be worth adding localization comments for the Text throughout the body of this property, although I guess since we don't have a framework-localized version of Text yet it'll have to change in the future anyway

ps2 replied:
Maybe we should just go and do this. It should be trivial.

mpangburn commented on DashKitUI/View Models/DashSettingsViewModel.swift:
```diff
@@ -104,6 +89,23 @@ class DashSettingsViewModel: DashSettingsViewModelProtocol {
+        }
+
+        pumpManager.suspendDelivery(withReminder: reminder) { (error) in
+            
```
[nit] Do you want a TODO here as well for displaying the error, if that work is for a separate PR?

ps2 replied:
Separate PR: https://tidepool.atlassian.net/browse/LOOP-1176

rickpasetto commented on DashKitUI/View Models/DashSettingsViewModel.swift:
```diff
@@ -147,34 +149,43 @@ extension DashPumpManager {
-        default:
-            absoluteRate = scheduledRate
+
+        if let tempBasal = state.unfinalizedTempBasal, !tempBasal.isFinished(at: Date()) {
```
Consider adding a unit test for this logic (I'm fine if this is in a later PR).

ps2 commented:
Based on new wording for the alerts, I also updated behavior to schedule the repeating alert immediately when suspend period ends. Also found a couple other bugs that I fixed. Appreciate a new review. Thanks.
